src/order/directed.lean

```diff
@@ -161,6 +162,72 @@ by assumption
 instance order_dual.is_directed_le [has_le α] [is_directed α (≥)] : is_directed αᵒᵈ (≤) :=
 by assumption
 
+section reflexive
+
+lemma directed_on.insert (h : reflexive r) (a : α) (s : set α) (hd : directed_on r s)
+(ha : ∀ b ∈ s, ∃ c ∈ s, a ≼ c ∧ b ≼ c) : directed_on r (insert a s) :=
```

Comment:
Is this true as an `iff` in some form?

Reply:
@eric-wieser The obvious statement would be:
```
lemma directed_on_insert_of_symmetric (hr : reflexive r) (hs : symmetric r) (a : α) (s : set α) :
  directed_on r (insert a s) →  directed_on r s ∧ ∀ b ∈ s, ∃ c ∈ s, a ≼ c ∧ b ≼ c := sorry
```
However, I don't think this is true. As a counter example, `directed_on (≤) (insert (1,1) ({(1,0), (0,1)} : set (ℕ × ℕ)))` is true:
```

example : directed_on (≤) (insert (1,1) ({(1,0), (0,1)} : set (ℕ × ℕ))) :=
begin
  rw directed_on,
  intros,
  cases H,
  { use x,
    split,
    { exact set.mem_union_left {(1, 0), (0, 1)} H, },
    { split,
      { exact rfl.ge, },
      { cases H_1,
        { rw [H, H_1], },
        cases H_1,
        { rw [H, H_1], norm_num, },
        { rw [set.mem_singleton_iff] at H_1, rw [H, H_1], norm_num, } } }, },
  { use (1,1),
    split,
    { simp, },
    split,
    { cases H,
      { rw H, norm_num, },
      { rw [set.mem_singleton_iff] at H, rw H, norm_num, }, },
    { cases H_1,
      { rw H_1, },
      { cases H_1,
        { rw H_1, norm_num, },
        { rw [set.mem_singleton_iff] at H_1, rw H_1, norm_num, }, }, }, }
end
```

But `(≤)` is not directed on `({(1,0), (0,1)} : set (ℕ × ℕ))`:
```
example : ¬directed_on (≤) ({(1,0), (0,1)} : set (ℕ × ℕ)) :=
begin
  rw directed_on,
  simp,
  use 1,
  use 0,
  split,
  { simp, },
  { use 0,
    use 1,
    split,
    { simp, },
    { intros x y h,
      norm_num,
      intro hx,
      have e1: ¬(x = 0) := by exact nat.one_le_iff_ne_zero.mp hx,
      have e2: ¬(x = 0 ∧ y = 1) := by exact not_and_of_not_left (y = 1) e1,
      have e2': (x = 0 ∧ y = 1) = false := by exact eq_false_intro e2,
      rw e2' at h,
      simp only [or_false] at h,
      exact h.2, } }
end
```